bool reverse
float64 x
float64 y
float64 z
float64 rotate_angle
float64 volume
float64 width